app/models/order.rb

```diff
@@ -2,5 +2,17 @@ class Order < ApplicationRecord
   belongs_to :user
   has_many :order_details, dependent: :destroy
 
+  before_save :update_subtotal
+
   enum status: { "init": 0, "processing": 1, "success": 2 }
+
+  def get_subtotal
+    order_details.collect { |oi| oi.valid? ? (oi.quantity * oi.unit_price) : 0 }.sum
```

Comment:
oi là gì thế em

Reply:
kieu bien phu em tao ra thay the cho order_detail de tinh tong thoi a